llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -1083,53 +1092,92 @@ bool LoopIdiomRecognize::processLoopStridedStore(
     return Changed;
 
   // Okay, everything looks good, insert the memset.
+  // MemsetArg is the number of bytes for the memset libcall, and the number
+  // of pattern repetitions if the memset.pattern intrinsic is being used.
+  Value *MemsetArg;
+  std::optional<int64_t> BytesWritten;
+
+  if (PatternValue && (HasMemsetPattern || ForceMemsetPatternIntrinsic)) {
+    const SCEV *TripCountS =
```

Comment:
There's a bunch of complexity in this section of the change that I'm not getting.  Since we need the NumBytes (in the original code) for the AAInfo, why not just leave that alone?  Isn't count (for memset.pattern intrinsic)  just the trip count of the loop by definition?

Reply:
I still think this might be reworked, but we're down to little enough code here I'm okay with that being a followup.

Reply:
Yes, it's the case of multiple stores in a loop iteration. I could perhaps build the necessary value with numbytes and a div, but I'm wary of cases where it may not be folded away.